### ISSUE TITLE
Gesture scrolling for touchscreens

### ISSUE THREAD

**Issue by s-ol** [VCVRack/Rack#362]
Feature request decoupled from #331; on touch screens it would be nice to be able to scroll the rack space without using the scroll bars. The proposal is to use a two-finger touch-and-pan to move around the viewport.

**Comment by SaveTheHuman5** [comment]
Hello Andrew, please check contact e-mail and check for e-mail with subject
 Controlerism and final solution for touch issues.
I think you like what you read!

**Comment by Uriel-A** [comment]
@AndrewBelt There's a good chance that touchscreen users will not have a keyboard handy in situations when the device is a convertible laptop or a Microsoft Surface without a keyboard attached so having keyboard modifier+touch is awkward at best.

What about https://github.com/mkalten/TUIO20_CPP or https://github.com/vialab/SMT ? Would it be possible to implement either of these into VCV?

**Comment by AndrewBelt** [comment]
@Uriel-A Those libraries are not quite related to the issue of adding touch support to a window launched with GLFW. If you'd like to contribute to an open source project, adding touch support to GLFW would be a great one that would help potentially thousands of other open source projects. https://github.com/glfw/glfw/issues/42

**Comment by Uriel-A** [comment]
@AndrewBelt I'm not a programmer!! 🤣 

Anyhow, how about a different idea?

Add an extra toggle button on top of VCV, let's say "Scroll". Click on it and any touch input would only translate as scrolling messages. Untoggle to resume normal operations.

Maybe might want to make that "Scroll" toggle button appear only via settings.json since only touchscreen users will need it.

**Comment by SaveTheHuman5** [comment]
Hello everybody.
A different idea is coming, with special focus in real usability for touch screen users, specially for all that one with no giant screens, 
can be used as stand alone or an as nice companion of your physical controllers, with instant access to all knobs.
https://www.youtube.com/watch?v=7o7NtwiyMwg

**Comment by nomchomsky** [comment]
Would be nice with touch input

**Comment by nomchomsky** [comment]
Yup....also found a fix for the knobs on the touchscreen with the json settings. Knob hold to false.
Vcvrack is really great on a large touchscreen. Feels like i suddenly have a huge eurorack system...where i never run out of cables🤪
